aicsimageio/transforms.py

```diff
@@ -105,59 +55,116 @@ def reshape_data(
         would have order "STCZYX"
 
     """
-    # Get operator
-    if isinstance(data, da.core.Array):
-        operator = da
-    else:
-        operator = np
+    # Check for parameter conflicts
+    for dim in Dimensions.DefaultOrderList:
+        # return_dims='TCZYX' and fixed dimensions 'C=1'
+        # Dimension is in kwargs
+        # Dimension is an integer
+        # Dimension is in return dimensions
+        if isinstance(kwargs.get(dim), int) and dim in return_dims:
+            raise ConflictingArgumentsError(
+                f"Argument return_dims={return_dims} and "
```

Comment:
I think the message could be even clearer for each case.  Tell the user something like "Selecting a single DIM index can not be returned in return_dims."